sdcm/utils/verify.py

```diff
@@ -0,0 +1,30 @@
+from difflib import unified_diff
+from io import StringIO
+from pathlib import Path
+
+import click
+
+
+class Fix(StringIO):
+    """A StringIO subclass that verifies the existing contents of a file and outputs the diff if there are changes."""
+    ERROR = False
+
+    def __init__(self, path, *args, **kwargs):
+        super().__init__("")
+        self.path = path
+
+    def close(self):
+        new = self.getvalue()
+        super().close()
+        old_path = Path(self.path)
+        if old_path.exists():
+            old = old_path.read_text()
+        else:
+            old = ""
+        diff = unified_diff(old.splitlines(keepends=True), new.splitlines(keepends=True), fromfile=str(self.path))
+        output = "".join(diff)
+        if len(output) > 0:
+            click.secho(output, fg="red")
+            self.__class__.ERROR = True
```

Comment:
communicating via the class back for the results ?

this pattern is confusing as heck...

Reply:
Changed, because the open() does not have mechanism to return back results, I had to create a "carrier" class which would be used to set error and translate them outside of function. It should be much cleaner (though still not perfect) now.